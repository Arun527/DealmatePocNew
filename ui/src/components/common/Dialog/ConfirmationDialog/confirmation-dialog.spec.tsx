import { fireEvent, render } from "@testing-library/react";
import TestContextProvider from "../../../../test/test-container";
import ConfirmationDialog from "./confirmation-dialog";

const getProps = (open = false) => ({
  open,
  handleAction: jest.fn(),
  actionLabels: ["Yes", "No"],
  outSideClick: true,
  width: "md",
  content: {
    header: "Confirmation",
    text: "Are you sure you want to proceed?",
  },
});

const renderApp = (props: any) => {
  return render(<ConfirmationDialog {...props} />, {
    wrapper: TestContextProvider,
  });
};

describe("ConfirmationDialog-Context", () => {
  let props, baseElement;

  props = getProps();
  ({ baseElement } = renderApp(props));

  afterEach(() => {
    jest.clearAllMocks();
    jest.resetAllMocks();
  });

  it("should match snapshot", () => {
    expect(baseElement).toMatchSnapshot();
  });

  it("should show the confirmation dialog and handle the 'Yes' action", async () => {
    props = getProps(true);
    const { baseElement } = renderApp(props);

    const message = baseElement.querySelectorAll('[data-pc-section="message"]');
    expect(message?.[0]?.textContent?.trim()).toBe(props.content.text);

    const buttons = baseElement.querySelectorAll(".p-dialog-footer button");
    expect(buttons[1].textContent?.trim()).toBe(props.actionLabels[0]);
    fireEvent.click(buttons[1]);
    expect(props.handleAction).toHaveBeenCalledWith("yes");
  });

  it("should close the confirmation dialog when 'No' is clicked", async () => {
    props = getProps(true);
    const { baseElement } = renderApp(props);

    const message = baseElement.querySelectorAll('[data-pc-section="message"]');
    expect(message?.[0]?.textContent?.trim()).toBe(props.content.text);

    const buttons = baseElement.querySelectorAll(".p-dialog-footer button");
    expect(buttons[0].textContent?.trim()).toBe(props.actionLabels[1]);
    fireEvent.click(buttons[0]);
    expect(props.handleAction).toHaveBeenCalledWith("no");
  });

  it("should close the dialog when close button is clicked", async () => {
    props = getProps(true);
    const { baseElement } = renderApp(props);
    const closeIcon = baseElement.querySelectorAll(
      '[data-pc-section="closebuttonicon"]'
    );
    fireEvent.click(closeIcon[0]);
    expect(props.handleAction).toHaveBeenCalledWith("close");
  });
});
