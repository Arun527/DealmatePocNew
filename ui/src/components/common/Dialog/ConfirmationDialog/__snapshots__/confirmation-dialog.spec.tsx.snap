// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationDialog-Context should match snapshot 1`] = `
<body>
  <div>
    <div />
  </div>
</body>
`;
